test/functional/wallet_rescan_unconfirmed.py

```diff
@@ -0,0 +1,97 @@
+
+        self.log.info("Create a parent tx and mine it in a block that will later be disconnected")
+        parent_amount = 100000
+        parent_address = w0.getnewaddress()
+        tx_parent_to_reorg = tester_wallet.send_to(
+            from_node=tester_node,
+            scriptPubKey=address_to_scriptpubkey(parent_address),
+            amount=parent_amount,
+        )
+        assert tx_parent_to_reorg["txid"] in tester_node.getrawmempool()
+        block_to_reorg = self.generate(tester_node, 1)[0]
+        assert_equal(len(tester_node.getrawmempool()), 0)
+
+        # Create an unconfirmed child transaction from the parent tx, sending all
+        # the funds to an unspendable address. Importantly, no change output is created so the
+        # transaction can't be recognized using its outputs. The wallet rescan needs to know the
+        # inputs of the transaction to detect it, so the parent must be processed before the child.
+        w0_utxos = w0.listunspent()
+
+        self.log.info("Create a child tx and wait for it to propagate to all mempools")
+        # The only UTXO available to spend is tx_parent_to_reorg.
+        assert_equal(len(w0_utxos), 1)
+        assert_equal(w0_utxos[0]["txid"], tx_parent_to_reorg["txid"])
+        tx_child_unconfirmed_sweep = w0.sendall([ADDRESS_BCRT1_UNSPENDABLE])
+        self.sync_all()
+        assert tx_child_unconfirmed_sweep["txid"] in tester_node.getrawmempool()
+        assert tx_child_unconfirmed_sweep["txid"] in import_wallet_node.getrawmempool()
+
+        self.log.info("Mock a reorg, causing parent to re-enter mempools after its child")
+        for node in self.nodes:
+            node.invalidateblock(block_to_reorg)
+            assert tx_parent_to_reorg["txid"] in node.getrawmempool()
+
+        self.log.info("Import descriptor wallet on another node")
+        descriptors_to_import = []
+        for item in w0.listdescriptors()["descriptors"]:
+            # descriptors_to_import.append(item)
+            # use timestamp 0 to tell wallet to rescan entire chain
+            if not item["internal"]:
+                descriptors_to_import.append({"desc": item["desc"], "timestamp": 0, "label": "w0 import"})
+        import_wallet_node.createwallet(wallet_name="w1", disable_private_keys=True, blank=True, descriptors=True)
+        w1 = import_wallet_node.get_wallet_rpc("w1")
+        w1.importdescriptors(descriptors_to_import)
+
+        self.log.info("Check that the importing node has properly rescanned mempool transactions")
+        # Check that parent address is correctly determined as ismine
+        test_address(w1, parent_address, solvable=True, ismine=True)
+        # This would raise a JSONRPCError if the transactions were not identified as belonging to the wallet.
+        assert_equal(w1.gettransaction(tx_parent_to_reorg["txid"])["confirmations"], 0)
+        assert_equal(w1.gettransaction(tx_child_unconfirmed_sweep["txid"])["confirmations"], 0)
```

Comment:
nit: I still think it's unnecessarily confusing for (current and future) reviewers to figure out why confirmations needs to equal 0 when afaiu this is completely unrelated to the test. I would either remove the value comparison or specifically state in the docs that we don't care about the `confirmations` value, just that the response is a JSON object with a field we recognize?

Reply:
We are testing that they are marked as in-mempool wallet transactions.

Reply:
Makes sense, thank you! I failed to appreciate that we're calling the wallet `gettransaction` RPC, and not the not `getrawtransaction` RPC.

Reply:
Ah yeah, it's a wallet-specific RPC. I've added a check that the "confirmations" value is 1 on wallet 0 a few lines up, prior to the reorg. Hopefully that also helps make it clear why we're testing this way.